doc/lag/Static Port Channel Design Document.md

```diff
@@ -0,0 +1,278 @@
+
+# 3 Design
+
+## 3.1 Overview
+Upon creating a static Port Channel, the "Teamd" daemon corresponding to that Port Channel is instantiated in static mode.
+
+### 3.1.1 Teamd configuration
+A user should be able to create the Port Channel in static or dynamic mode. For static mode, the CONFIG_DB is updated with a "on": "true".
+Note: Port Channels are dynamic by default in SONiC. 
+
+### 3.1.2 Team Manager
+Team Manager listens to Port Channel update events from the PORTCHANNEL and PORTCHANNEL_MEMBER table in CONFIG_DB. For the Static Port Channel add event it starts Teamd  with round-robin as the runner; whereas for the dynamic Port Channel Teammgr starts Teamd with lacp as the runner. 
+
+When static port-channel is configured then teammgr invokes teamd instance as follows
+root          27       1  0 Jun17 ?        00:01:10 /usr/bin/teamd -r -t PortChannel0001 -c {"device":"PortChannel0001","hwaddr":"00:e0:ec:7a:88:11","runner":{"name":"roundrobin","min_ports":1}} -L /var/warmboot/teamd/ -g -d
+
+round-robin load-balancing is configured for the kernel interface port-channel.
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+To support the static mode for a Port Channel, the PORTCHANNEL table is modified to add a new key-value pair where the value is a "true" or a "false".
+```
+"PORTCHANNEL|PortChannel0001": {
+        "type": "hash",
+        "value": {
+            "on": "true"
+        }
+    }
+
+127.0.0.1:6379[4]> hgetall "PORTCHANNEL|PortChannel0001"
+ 9) "on"
+10) "true"
+
+```
+### 3.2.2 APP DB
+No new tables are needed.
+
+### 3.2.3 STATE DB
+No new tables are needed.
+
+### 3.2.4 ASIC DB
+No new tables are needed.
+
+### 3.2.5 COUNTER DB
+No new tables are needed.
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+No changes are needed.
+
+### 3.3.2 Other Process
+No changes are needed.
+
+## 3.4 SyncD
+No changes are needed.
+
+## 3.5 SAI
+No changes are needed.
+
+## 3.6 CLI
+
+### 3.6.1 Data Models
+Please refer [Openconfig support for Port Channel](#openconfig-support-for-port-channel) for details on the Data Model.
+
+### 3.6.2 Configuration Commands
+
+Please refer [Openconfig support for Port Channel](#openconfig-support-for-port-channel) for details on KLISH CLI.
+
+```
+click CLI
+1. config portchannel add PortChannel<number> --static=true
+   Create a Static Port Channel.
+
+```
+
+### 3.6.3 Show Commands
+
+Please refer [Openconfig support for Port Channel](#openconfig-support-for-port-channel) for details on KLISH CLI.
+
+```
+click CLI
+1. show interfaces portchannel
+This command is modified to show the Port Channel Protocol as None for a Static Port Channel.
+
+Flags: A - active, I - inactive, Up - up, Dw - Down, N/A - not available,
+       S - selected, D - deselected, * - not synced
+  No.  Team Dev         Protocol       Ports
+-----  ---------------  -------------  --------------
+ 0001  PortChannel0001  STATIC(A)(Up)  Ethernet112(S)
```

Comment:
Typically the 'Protocol' column field specifies protocol used for the portchannel.

For Dynamic portchannel, 'LACP' keyword is good.
For Static portchannel, 'NONE' keyword sounds more logical and would be inline with NOS vendor Cisco. 
Note: I see that Arista  uses 'Static' keyword though.

Reply:
Done